// @fullcalendar/resource-daygrid/index.js
import { createPlugin } from '@fullcalendar/core';
import premiumCommonPlugin from '@fullcalendar/premium-common';
import resourcePlugin from '@fullcalendar/resource';
import dayGridPlugin from '@fullcalendar/daygrid';
import { ResourceDayTableView } from './internal.js';
import '@fullcalendar/core/internal';
import '@fullcalendar/core/preact';
import '@fullcalendar/daygrid/internal';
import '@fullcalendar/resource/internal';

var index = createPlugin({
    name: '@fullcalendar/resource-daygrid',
    premiumReleaseDate: '2023-02-07',
    deps: [
        premiumCommonPlugin,
        resourcePlugin,
        dayGridPlugin,
    ],
    initialView: 'resourceDayGridDay',
    views: {
        resourceDayGrid: {
            type: 'dayGrid',
            component: ResourceDayTableView,
            needsResourceData: true,
        },
        resourceDayGridDay: {
            type: 'resourceDayGrid',
            duration: { days: 1 },
        },
        resourceDayGridWeek: {
            type: 'resourceDayGrid',
            duration: { weeks: 1 },
        },
        resourceDayGridMonth: {
            type: 'resourceDayGrid',
            duration: { months: 1 },
            fixedWeekCount: true,
        },
    },
});

export { index as default };
